### PR TITLE
added message

### PR DESCRIPTION
**Pull request by ritwika314**
Minor restructuring, added message types and messages. One base_tests class file testing access methods for everything. Please review and accept.

### CODE REVIEW

File: src/functionality/comm_funcs.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    :type port: int
+
+    :param retry: number of re-sends of the message
+    :type retry: int
+
+    TODO: USE different serialization .
+    TODO: write test
+    """
+
+    client_sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
```
Comment (hc825b) [inline src/functionality/comm_funcs.py]: Use `with` statement to create and close `client_sock` automatically. This way it also handles some default exceptions.

File: src/objects/message.py [hunk 2]
```diff
@@ -39,67 +45,39 @@ def __eq__(self, other):
...
-        getter method for message content
-        :return: message contents
-        """
         return self.__content
 
+    @property
+    def message_type(self) -> MsgType:
+        return self.__message_type
+
+    @sender.setter
```
Comment (hc825b) [inline src/objects/message.py]: Why `sender`, `content`, and `message_type` requires setter methods?

**Review by hc825b — CHANGES_REQUESTED** [review]
See comments in each file.

Also missing `src.motion.planners.simpleplanner` module

**Comment by ritwika314** [comment]
removed simpleplanner as it is useless. removed all tests as well.

File: src/objects/message.py (hunk 2 above)
Comment (ritwika314) [reply src/objects/message.py]: doesn't. youre right.

File: src/functionality/comm_funcs.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    """
+
+    with socket.socket(socket.AF_INET, socket.SOCK_DGRAM) as client_sock:
+        client_sock.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)
+        for i in range(retry):
+            a = pickle.dumps(msg)
+            if len(a) > MAX_UDP_SIZE:
+                raise ValueError('Message too large')
+            client_sock.sendto(a, (ip, port))
+        client_sock.close()
```
Comment (hc825b) [inline src/functionality/comm_funcs.py]: Redundant `close()`

File: src/config/configs.py [hunk 4]
```diff
@@ -238,7 +237,7 @@ class MoatConfig(object):
 
     def __init__(self, waypoint_topic: str, reached_topic: str, rospy_node: str, bot_name: str, queue_size: int,
                  bot_type: BotType, pos_node: str, pos_msg_type=None, rchd_msg_type=None,
-                 planner: Planner = SimplePlanner()):
+                 planner: tp.Union[Planner, None] = None):
```
Comment (hc825b) [inline src/config/configs.py]: Use `Optional[T]` instead of `Union[T, None]` for short

**Review by hc825b — CHANGES_REQUESTED** [review]
Some more minor comments.
See comments in each file.

**Review by hc825b — APPROVED** [review]

**Pull request merged**